foam_controlwrapper/tests/test_foam_interface.py

```diff
@@ -0,0 +1,292 @@
+        self.cellPoints = []
+        for cell in self.mesh.iter_cells():
+            self.cellPoints.append(len(cell.points))
+            for puid in cell.points:
+                self.cellPoints.append(self._uuidToFoamLabel[puid])
+
+        # make patch information
+        self.patchNames = []
+        self.patchFaces = []
+        for patchName in patchNameFacesMap:
+            self.patchNames.append(patchName)
+            self.patchFaces.append(len(patchNameFacesMap[patchName]))
+            for face in patchNameFacesMap[patchName]:
+                self.patchFaces.append(face)
+
+        if not self.patchNames:
+            error_str = 'Could not initialize with mesh  {}. '
+            error_str += 'Mesh has not boundary face definitions.'
+            raise ValueError(error_str.format(self.mesh.name))
+
+        # this to have controlDict file for mesh definition
+        FoamFiles().write_default_files(self.path, 'simpleFoam',
+                                        self.time, False)
+
+        foamface.init(self.name, os.path.abspath(os.path.join(self.path,
+                                                              os.pardir)))
+
+    def tearDown(self):
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+
+    def test_add_mesh(self):
+        """Test addMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+
+    def test_write_mesh(self):
+        """Test writeMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        foamface.writeMesh(self.name)
+
+    def test_get_point_coordinates(self):
+        """Test getPointCoordinates method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        for point in self.points:
+            coords = foamface.getPointCoordinates(
+                self.name,
+                self._uuidToFoamLabel[point.uid])
+            self.assertEqual(point.coordinates, tuple(coords))
+
+    def test_get_face_points(self):
+        """Test getFacePoints method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        for face in self.faces:
+            pointLabels = foamface.getFacePoints(
+                self.name,
+                self._uuidToFoamLabel[face.uid])
+            puids = [self._foamPointLabelToUuid[lbl] for lbl in pointLabels]
+            self.assertEqual(puids, face.points)
+
+    def test_get_cell_points(self):
```

Comment:
There should be a check where cells have Data since the foam_interface supports data attributes for cells correct?